### PR TITLE
Public API Doc updates for Delivered roles

### PR DESCRIPTION
**Pull request by alifatolahi-okta**
## Description:
- **What's changed?** Roles API reference and concpets docs updated to support delivered roles concepts as well as the two standard roles for IGA
- **Is this PR related to a Monolith release?** `2022.08.0`

### Resolves:

* [OKTA-492535](https://oktainc.atlassian.net/browse/OKTA-492535)

### CODE REVIEW

**Review by macyabbey-okta — COMMENTED** [review]
Super comprehensive and good doc! Left a couple notes. Don't think I know enough here to be the approver but it all lines up with what I know.

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (ryanbradley-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:71]: +1 I didn't follow this either.

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md
Comment (ryanbradley-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:3197]: Pluralization of "manage access request**s**"

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md
Comment (ryanbradley-okta) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:3257]: Same question here as to whether these should be "Access Certifications" and "Access Requests" to match the product or whether they should match the ORN names

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (rogertang-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:34]: We had such a discussion on this topic before.

So, we have two versions of label for standard roles:
- Label in built-in version. This is the value for the `name` column in `CustomeRoles` table. It has to be unique. It's not the text shown in UI which is the label in localized version. Adding an OKTA prefix to build-in version is to avoid naming conflict with customer-created role.
- Label in localized version. The label showing up in UI will be translated into localized text. Even for english language locale, the label in built-in version is NOT used directly as the localized label. I assume the localized label has no OKTA prefix and if it does then we need a bug jira. 

@alifatolahi-okta, I feel like the label in built-in version is more for internal use. It sounds like it's not a good idea to expose that to the end user. thoughts?

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (rogertang-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:71]: Maybe add some example here to illustrate what's standard role `id` and `type`?

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md]: thanks, fixed

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md]: We have examples in the reference doc, I have re-worded this section, hopefully it's more clear now.

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:34]: There is no localization for standard role names right now, for existing OOTB roles also we simply surface the DB label in public API. If these names have to change, and it's your call really, then it will have to change in code too.

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md]: great call, change

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:188]: fixed

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md]: fixed

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/reference/api/roles/index.md:3099]: ditto same response :)

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (macyabbey-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:34]: +1 for removing "OKTA" label...found it jarring - is this a problem for our release timelines @alifatolahi-okta / @rogertang-okta or it could be addressed as bug in July?

Comment (rogertang-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:34]: I created a JIRA for that. Please let me know we want to proceed it. 

https://oktainc.atlassian.net/browse/OKTA-509517

Comment (ryanbradley-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:34]: @rogertang-okta I think that we should proceed with this change. I understand the risk of collision if a customer already has created a custom admin role with this name. However, seeing as none of these capabilities are included in the IAM Framework today, I think that this risk is very low.

**Review by rogertang-okta — APPROVED** [review]

**Review by ryanbradley-okta — APPROVED** [review]
Looks good to me

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md
Comment (KalAdlakha) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:3264]: Resources should be plural.

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md
Comment (KalAdlakha) [inline packages/@okta/vuepress-site/docs/reference/api/roles/index.md:3200]: This instead of these.

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md]: fixed

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/reference/api/roles/index.md]: fixed to the latest label

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/reference/api/roles/index.md]: fixed

File: packages/@okta/vuepress-site/docs/reference/api/roles/index.md
Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/reference/api/roles/index.md]: fixed

**Review by vasishtajayanti-okta — APPROVED** [review]

**Review by bhamel-okta — APPROVED** [review]

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (bhamel-okta) [inline packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:75]: @ryanbradley-okta I know you advocated to remove 'OKTA_' from the standard IAM roles. There is a non-zero chance we'll show the resource sets here in the UI in the future, would you want OKTA_ removed from them as well if/when they are displayed to customers?

Comment (ryanbradley-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:75]: @bhamel-okta Yes, we should remove the 'OKTA_' prefix if this may be seen by customers.

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (macyabbey-okta) [inline packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:71]: nit: `pre-defined` in 1. and `predefined` in 2. Apparently they are both in the dictionary but couldn't help myself from commenting.

Comment (alifatolahi-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:71]: Ah good catch. @brianduffield-okta do you have any preference which to use?

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (rogertang-okta) [inline packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:75]: I'm also renaming the role enum names. U may want to uptake that too.

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (vasishtajayanti-okta) [inline packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:34]: @rogertang-okta can you confirm if "Access Certifications Admin" needs to be changed to "Access Certifications Administrator" and same for "Access Requests Admin" -> "Access Requests Administrator"?

Comment (rogertang-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:34]: yes confirmed

File: packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md
Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:71]: @alifatolahi-okta we don't have an internal style. In these cases we defer to the Microsoft Manual of style: https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/pre. So, in this case, I'd suggest `predefined`.

Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/concepts/role-assignment/index.md:71]: @alifatolahi-okta FYI, I can make this minor update when merging the release branch into our main branch later this evening. No need to update here.

**Pull request merged**
